llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -4278,8 +4287,18 @@ bool LoopVectorizationCostModel::isScalarWithPredication(
   switch(I->getOpcode()) {
   default:
     return true;
-  case Instruction::Call:
-    return !VFDatabase::hasMaskedVariant(*(cast<CallInst>(I)), VF);
+  case Instruction::Call: {
+    if (!isa<PrefetchInst>(I))
+      return !VFDatabase::hasMaskedVariant(*(cast<CallInst>(I)), VF);
+    auto *Ptr = getPrefetchPointerOperand(I);
```

Comment:
Can't you implement `hasMaskedVariant` for prefetch?

Reply:
Thanks for the comment.

I think I can't implement it. This is because `HasMaskedVariant` checks if the library you are using has a masked and vectorized implementation. I don't think prefetch will ever be implemented in the library. This is because prefetch intrinsic is converted to a single instruction. Therefore, we did not consider using HasMaskedVariant.